### PR TITLE
Docs: Enterprise Vault feature

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
Adds documentation for the Grafana Enterprise 7.1 Vault integration.

**Special notes for your reviewer**:
Replaces #25861 as Emil is away and I don't have write access to his fork.

### CODE REVIEW

File: docs/sources/administration/configuration.md [hunk 1]
```diff
@@ -86,12 +86,14 @@ export GF_PLUGIN_GRAFANA_IMAGE_RENDERER_RENDERING_IGNORE_HTTPS_ERRORS=true
 
 > Only available in Grafana 7.1+.
 
+> For any changes to `conf/grafana.ini` (or corresponding environment variables) to take effect, you must restart Grafana for the changes to take effect.
```
Comment (xlson) [inline docs/sources/administration/configuration.md:89]:
```suggestion
> For any changes to `conf/grafana.ini` (or corresponding environment variables) to take effect, you must restart Grafana.
```

**Review by xlson — APPROVED** [review]
Great work! Clear and easy to understand.

File: docs/sources/administration/configuration.md [hunk 2]
```diff
@@ -86,12 +86,14 @@ export GF_PLUGIN_GRAFANA_IMAGE_RENDERER_RENDERING_IGNORE_HTTPS_ERRORS=true
...
+
 If any of your options contains the expression `$__<provider>{<argument>}`
 or `${<environment variable>}`, then they will be processed by Grafana's
 variable expander. The expander runs the provider with the provided argument
 to get the final value of the option.
 
-There are two providers: `env` and `file`.
+There are three providers: `env`, `file` and `vault`.
```
Comment (oddlittlebird) [inline docs/sources/administration/configuration.md:96]:
```suggestion
There are three providers: `env`, `file`, and `vault`.
```

File: docs/sources/enterprise/_index.md [hunk 3]
```diff
@@ -54,6 +54,7 @@ With Grafana Enterprise, you get access to new features, including:
 * [Export dashboard as PDF]({{< relref "export-pdf.md" >}})
 * [White labeling]({{< relref "white-labeling.md" >}}) to customize Grafana from the brand and logo to the footer links.
 * [Usage insights]({{< relref "usage-insights.md" >}}) to understand how your Grafana instance is used.
+* [Vault integration]({{< relref "vault.md" >}}) to manage your configuration or provisionning secrets with Vault.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/_index.md:57]:
```suggestion
* [Vault integration]({{< relref "vault.md" >}}) to manage your configuration or provisioning secrets with Vault.
```

File: docs/sources/enterprise/enterprise-configuration.md [hunk 4]
```diff
@@ -186,3 +186,33 @@ List of comma- or space-separated roles that will be mapped to the Admin role.
...
+
+## [keystore.vault]
+
+### url
+
+Location of the Vault server.
+
+### namespace
+
+Vault's namespace if using Vault with multi-tenancy.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/enterprise-configuration.md:198]:
```suggestion
Vault namespace if using Vault with multi-tenancy.
```

File: docs/sources/enterprise/enterprise-configuration.md [hunk 5]
```diff
@@ -186,3 +186,33 @@ List of comma- or space-separated roles that will be mapped to the Admin role.
...
+
+### auth_method
+
+Method for authenticating towards Vault. Vault is inactive if this option is not set. Current possible values: `token`.
+
+### token
+
+ Secret token to connect to Vault when auth_method is `token`.
+
+ ### lease_renewal_interval
```
Comment (oddlittlebird) [inline docs/sources/enterprise/enterprise-configuration.md:208]: Looks like this h3 and the next two have a space in front, invalidating the heading. Please correct.

File: docs/sources/enterprise/vault.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+[menu.docs]
+parent = "enterprise"
+weight = 700
++++
+
+# Vault integration
+
+> Only available in Grafana Enterprise v7.1+.
+
+For organizations managing their secrets with [Hashicorp Vault](https://www.hashicorp.com/products/vault),
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:15]: This sentence does not make sense. Consider making it two sentences.

File: docs/sources/enterprise/vault.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+Grafana integrates with Vault for [Configuration]({{< relref "../administration/configuration.md" >}})
+and [Provisioning]({{< relref "../administration/provisioning.md" >}}).
+
+> Note: If you have Grafana set up for [high availability]({{< relref "../tutorials/ha_setup.md" >}}) we advise not to use dynamic secrets for provisioning files.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:19]:
```suggestion
> **Note:** If you have Grafana set up for [high availability]({{< relref "../tutorials/ha_setup.md" >}}), then we advise not to use dynamic secrets for provisioning files.
```

File: docs/sources/enterprise/vault.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+> Each Grafana instance is responsible for renewing its own leases. Your data sources' leases might expire when one of your Grafana servers shuts down.
+
+## Configuration
+
+Before using Vault you need to activate it by providing a URL, authentication method (currently only token),
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:24]:
```suggestion
Before using Vault, you need to activate it by providing a URL, authentication method (currently only token),
```

File: docs/sources/enterprise/vault.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+and a token for your Vault service. Grafana will automatically renew the service token if it is renewable and
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:25]:
```suggestion
and a token for your Vault service. Grafana automatically renews the service token if it is renewable and
```

File: docs/sources/enterprise/vault.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+set up with a limited lifetime.
+
+If you're using short-lived leases, you can also configure how often Grafana should renew the lease and for
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:28]:
```suggestion
If you're using short-lived leases, then you can also configure how often Grafana should renew the lease and for
```

File: docs/sources/enterprise/vault.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+how long. We recommend keeping the defaults here unless you run into problems.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:29]:
```suggestion
how long. We recommend keeping the defaults unless you run into problems.
```

File: docs/sources/enterprise/vault.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+Example for `vault server -dev`:
+
+```ini
+[keystore.vault]
+url = http://127.0.0.1:8200 # HTTP should only be used for local testing
+auth_method = token
+token = s.sAZLyI0r7sFLMPq6MWtoOhAN # replace with your key
+```
+
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:58]:
```suggestion
```

File: docs/sources/enterprise/vault.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+
+## Using the Vault expander
+
+When you've configured Vault you'll need to set the configuration or provisioning files you wish to
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:62]:
```suggestion
After you configure Vault, you must set the configuration or provisioning files you wish to
```

File: docs/sources/enterprise/vault.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+use Vault. Vault configuration is an extension of configuration's [variable expansion]({{< relref "../administration/configuration.md#variable-expansion" >}}) and follows the
+`$__vault{<argument>}` syntax.
+
+The argument to Vault consists of three parts separated by a colon: the first part specifies which secrets engine
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:66]: Make the three parts a bulleted list, each part with a separate bullet and treated like a separate sentence.

File: docs/sources/enterprise/vault.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+Grafana supports Vault's [K/V version 2](https://www.vaultproject.io/docs/secrets/kv/kv-v2) storage engine which
+is used to store and retrieve arbitrary secrets as `kv`. 
+
+```ini
+$__vault{kv:secret/grafana/smtp:username}
+```
+
+#### Databases
+
+Vault's [Databases secrets engines](https://www.vaultproject.io/docs/secrets/databases) is a family of
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:87]:
```suggestion
The Vault [databases secrets engines](https://www.vaultproject.io/docs/secrets/databases) is a family of
```

File: docs/sources/enterprise/vault.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+secret engines which shares a similar syntax and grants the user dynamic access to a database.
+You can use this both for setting up Grafana's own database access and for provisioning data sources.
+
+```ini
+$__vault{database:database/creds/grafana:username}
+```
+
+### Examples
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:95]: Please add a brief intro to this section, explaining to the reader what the examples are of and how to use them.

File: docs/sources/enterprise/vault.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+name = grafana
+user = $__vault{database:database/creds/grafana:username}
+password = $__vault{database:database/creds/grafana:password}
+```
+
+#### Provisioning
+
+The following is a full examples of a provisioning YAML file setting up a MySQL data source using Vault's
+database secrets engine.
+Refer to the [Provisioning]({{< relref "../administration/provisioning.md" >}}) section for more information.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:121]:
```suggestion
Refer to [Provisioning]({{< relref "../administration/provisioning.md" >}}) for more information.
```

File: docs/sources/enterprise/vault.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+```ini
+$__vault{database:database/creds/grafana:username}
+```
+
+### Examples
+
+#### Configuration
+
+The following is a partial example for using Vault to set up a Grafana configuration file's email and database credentials.
+Refer to the [Configuration]({{< relref "../administration/configuration.md" >}}) section for more information.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:100]:
```suggestion
Refer to [Configuration]({{< relref "../administration/configuration.md" >}}) for more information.
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Documentation edits

File: docs/sources/administration/configuration.md (hunk 1 above)
Comment (oddlittlebird) [reply docs/sources/administration/configuration.md:89]: I agree with this edit.

File: docs/sources/enterprise/vault.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+[menu.docs]
+parent = "enterprise"
+weight = 700
++++
+
+# Vault integration
+
+> Only available in Grafana Enterprise v7.1+.
+
+If you are managing your secrets with [Hashicorp Vault](https://www.hashicorp.com/products/vault), you can use them for [Configuration]({{< relref "../administration/configuration.md" >}})
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:15]:
```suggestion
If you manage your secrets with [Hashicorp Vault](https://www.hashicorp.com/products/vault), you can use them for [Configuration]({{< relref "../administration/configuration.md" >}})
```

File: docs/sources/enterprise/vault.md [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+and [Provisioning]({{< relref "../administration/provisioning.md" >}}).
+
+> **Note:** If you have Grafana set up for [high availability]({{< relref "../tutorials/ha_setup.md" >}}), then we advise not to use dynamic secrets for provisioning files.
+> Each Grafana instance is responsible for renewing its own leases. Your data sources' leases might expire when one of your Grafana servers shuts down.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/vault.md:19]:
```suggestion
> Each Grafana instance is responsible for renewing its own leases. Your data source leases might expire when one of your Grafana servers shuts down.
```

**Review by oddlittlebird — APPROVED** [review]
A couple minor edits, then good to go!

**Pull request merged**